src/library/scala/concurrent/package.scala

```diff
@@ -140,17 +159,20 @@ package concurrent {
   /**
    * `Await` is what is used to ensure proper handling of blocking for `Awaitable` instances.
    *
-   * While occasionally useful, e.g. for testing, it is recommended that you avoid Await
-   * when possible in favor of callbacks and combinators like onComplete and use in
-   * for comprehensions. Await will block the thread on which it runs, and could cause
-   * performance and deadlock issues.
+   * While occasionally useful, e.g. for testing, it is recommended that you avoid Await whenever possible—
+   * instead favoring combinators and/or callbacks.
+   * Await's `result` and `ready` methods will block the calling thread's execution until they return,
+   * which will cause performance degradation, and possibly, deadlock issues.
    */
   object Await {
     /**
      * Await the "completed" state of an `Awaitable`.
      *
      * Although this method is blocking, the internal use of [[scala.concurrent.blocking blocking]] ensures that
-     * the underlying [[ExecutionContext]] is prepared to properly manage the blocking.
+     * the underlying [[ExecutionContext]] is given an opportunity to properly manage the blocking.
+     *
+     * WARNING: It is strongly discouraged to supply lengthy timeouts since the progress of the calling thread will be
+     * suspended—blocked—until either the `Awaitable` becomes ready or the timeout expires.
```

Comment:
`suspended—blocked—until`

Does this mean "..will be suspended and therefore blocked until either .."